action.yml

```diff
@@ -251,54 +268,107 @@ runs:
+      gha_target_python_version = os.environ['GHA_TARGET_PYTHON_VERSION']
+      gha_testing_type = os.environ['GHA_TESTING_TYPE']
 
       # Validate GHA inputs
-      if coverage not in {'always', 'never', 'auto'}:
+      if gha_coverage not in {'always', 'never', 'auto'}:
           print(
               '::error ::`coverage` must have one of the values `always`,'
-              f' `never`, or `auto`.  The current value is `{coverage}`.'
+              f' `never`, or `auto`.  The current value is `{gha_coverage}`.'
           )
           sys.exit(1)
 
+      # Compose ansible-test arguments
+      command = [gha_testing_type, '-v', '--color']
+
       # Compute coverage and change detection arguments
-      coverage_arg = '' if coverage == 'never' else '--coverage'
-      change_detection_arg = ''
-      if pull_request_branch and pull_request_change_detection:
-          if coverage == 'auto':
+      has_change_detection = (
+          gha_pull_request_branch and gha_pull_request_change_detection
+      )
+      if has_change_detection:
+          if gha_coverage == 'auto':
               print(
                   'Disabling coverage reporting due to pull request '
                   'change detection being enabled.',
               )
-              coverage_arg = ''
-          change_detection_arg = (
-              f'--changed --base-branch {pull_request_branch}'
-          )
-
-      # Determine integration test flags
-      integration_flags = []
-      if integration_retry_on_error:
-          integration_flags.append('--retry-on-error')
-      if integration_continue_on_error:
-          integration_flags.append('--continue-on-error')
-      if integration_diff:
-          integration_flags.append('--diff')
-
-      # Set computed coverage-arg and change-detection-arg
-      set_output('coverage-arg', coverage_arg)
-      set_output('change-detection-arg', change_detection_arg)
-      set_output('integration-flags', ' '.join(integration_flags))
+              gha_coverage = 'never'
+          command.extend([
+              '--changed',
+              '--base-branch',
+              gha_pull_request_branch,
+          ])
+      has_coverage = gha_coverage != 'never'
+      if has_coverage:
+          command.append('--coverage')
+
+      if gha_testing_type == 'sanity':
+          command.append('--junit')
+
+      if gha_testing_type == 'integration':
+          if gha_integration_retry_on_error:
+              command.append('--retry-on-error')
+          if gha_integration_continue_on_error:
+              command.append('--continue-on-error')
+          if gha_integration_diff:
+              command.append('--diff')
+          if gha_container_network != '':
+              command.append(f"--docker-network={gha_container_network}")
+
+      if (
+          gha_testing_type == 'integration'
+          and gha_ansible_core_version not in {
+              'stable-2.9', 'stable-2.10', 'stable-2.11'
```

Comment:
In follow-up refactorings, it would be nice to put such sets into vars as well, to label what this piece of data means.